aodndata/soop/soop_tmv_nrt.py

```diff
@@ -0,0 +1,154 @@
+    date_format = '%Y/%m/%d %H:%M:%S'
+    df['TIME'] = pd.to_datetime(df['TIME'], format=date_format)
+
+    df = df[np.isfinite(df['LATITUDE'])]
+    df = df[np.isfinite(df['LONGITUDE'])]
+
+    # force make time monotonic and time not duplicated
+    df.set_index('TIME', inplace=True)
+    df.sort_index(axis=0, inplace=True)
+    df = df[~df.index.duplicated(keep='first')]
+    return df
+
+
+def netcdf_writer(log_path, output_dir):
+    df = parse_log_file(log_path)
+    ship_callsign_ls = ship_callsign_list()
+    log_filename = os.path.basename(log_path)
+
+    if SHIP_CODE not in ship_callsign_ls:
+        raise ValueError(
+            "Missing vessel callsign in file name '{name}'.".format(name=log_filename))
+
+    if SOOP_NRT_LOG_PATTERN.match(log_filename):
+        fields = get_pattern_subgroups_from_string(log_filename, SOOP_NRT_LOG_PATTERN)
+        product_code = fields['product_code']
+
+    if product_code in ['D2M', 'M2D']:
+        product_type = "transect"
+        featureType = "trajectory"
+        template = DatasetTemplate.from_json(NC_JSON_TEMPLATE_TRAJECTORY)
+    else:
+        product_type = "mooring"
+        featureType = "timeSeries"
+        template = DatasetTemplate.from_json(NC_JSON_TEMPLATE_MOORING)
+
+    template.global_attributes.update({
+        'product_type': product_type})
+
+    time_val_dateobj = date2num(df.index.to_pydatetime(),
+                                template.variables['TIME']['units'],
+                                template.variables['TIME']['calendar'])
+
+    # replace all nan with FillValue from template value
+    df.replace(np.nan, template.variables['LATITUDE']['_FillValue'], inplace=True)
+
+    template.variables['TIME']['_data'] = time_val_dateobj
+    template.variables['LATITUDE']['_data'] = df.LATITUDE.values
+    template.variables['LONGITUDE']['_data'] = df.LONGITUDE.values
+
+    template.variables['TEMP']['_data'] = df.TEMP.values
+    template.variables['PSAL']['_data'] = df.PSAL.values
+    template.variables['TURB']['_data'] = df.TURB.values
+    template.variables['CPHL']['_data'] = df.CPHL.values
+
+    if '1SecRaw' in log_filename:
+        time_period = '1sec'
+    else:
+        time_period = '10secs'
+
+    template.global_attributes.update({
+        'time_coverage_start': df.index.strftime('%Y-%m-%dT%H:%M:%SZ')[0],
+        'time_coverage_end': df.index.strftime('%Y-%m-%dT%H:%M:%SZ')[-1],
+        'featureType': featureType,
+        'date_created': pd.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ"),
+        'platform_code': SHIP_CODE,
+        'vessel_name': ship_callsign_ls[SHIP_CODE],
+        'geospatial_lat_min': df.LATITUDE.dropna().min(),
+        'geospatial_lat_max': df.LATITUDE.dropna().max(),
+        'geospatial_lon_min': df.LONGITUDE.dropna().min(),
+        'geospatial_lon_max': df.LONGITUDE.dropna().max(),
+        'time_period': time_period,
+        'history': "File created %s" % pd.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
```

Comment:
Should standardise the string formatting style to use the ".format" style rather than a mix of multiple styles